examples/documentation_examples/examples.go

```diff
@@ -2097,3 +2097,645 @@ func ChangeStreamExamples(t *testing.T, db *mongo.Database) {
+					},
+					bson.D{
+						{"fruit", "apple"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+				}},
+			},
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-01-11T07:15:00.000Z"},
+				}},
+				{"items", bson.A{
+					bson.D{
+						{"fruit", "banana"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+					bson.D{
+						{"fruit", "apple"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+					bson.D{
+						{"fruit", "papaya"},
+						{"quantity", 3},
+						{"price", 4.0},
+					},
+				}},
+			},
+		}
+
+		result, err := salesColl.InsertMany(ctx, docs)
+
+		// End Aggregation Example 1
+
+		require.NoError(t, err)
+		require.Len(t, result.InsertedIDs, 6)
+	}
+	{
+		// Start Aggregation Example 2
+		pipeline := mongo.Pipeline{
+			{
+				{"$match", bson.D{
+					{"items.fruit", "banana"},
+				}},
+			},
+			{
+				{"$sort", bson.D{
+					{"date", 1},
+				}},
+			},
+		}
+
+		cursor, err := salesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 2
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+
+		ok := cursor.Next(ctx)
+
+		require.True(t, ok)
+	}
+	{
+		// Start Aggregation Example 3
+		pipeline := mongo.Pipeline{
+			{
+				{"$unwind", "$items"},
+			},
+			{
+				{"$match", bson.D{
+					{"items.fruit", "banana"},
+				}},
+			},
+			{
+				{"$group", bson.D{
+					{"_id", bson.D{
+						{"day", bson.D{
+							{"$dayOfWeek", "$date"},
+						}},
+					}},
+					{"count", bson.D{
+						{"$sum", "$items.quantity"},
+					}},
+				}},
+			},
+			{
+				{"$project", bson.D{
+					{"dayOfWeek", "$_id.day"},
+					{"numberSold", "$count"},
+					{"_id", 0},
+				}},
+			},
+			{
+				{"$sort", bson.D{
+					{"numberSold", 1},
+				}},
+			},
+		}
+
+		cursor, err := salesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 3
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+
+		ok := cursor.Next(ctx)
+
+		require.True(t, ok)
+	}
+	{
+		// Start Aggregation Example 4
+		pipeline := mongo.Pipeline{
+			{
+				{"$unwind", "$items"},
+			},
+			{
+				{"$group", bson.D{
+					{"_id", bson.D{
+						{"day", bson.D{
+							{"$dayOfWeek", "$date"},
+						}},
+					}},
+					{"items_sold", bson.D{
+						{"$sum", "$items.quantity"},
+					}},
+					{"revenue", bson.D{
+						{"$sum", bson.D{
+							{"$multiply", bson.A{"$items.quantity", "$items.price"}},
+						}},
+					}},
+				}},
+			},
+			{
+				{"$project", bson.D{
+					{"day", "$_id.day"},
+					{"revenue", 1},
+					{"items_sold", 1},
+					{"discount", bson.D{
+						{"$cond", bson.D{
+							{"if", bson.D{
+								{"$lte", bson.A{"$revenue", 250}},
+								{"then", 25},
+								{"else", 0},
+							}},
+						}},
+					}},
+				}},
+			},
+		}
+
+		cursor, err := salesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 4
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+
+		ok := cursor.Next(ctx)
+
+		require.True(t, ok)
+	}
+	{
+		// Start Aggregation Example 5
+		docs := []interface{}{
```

Comment:
Would it be possible to insert the test data for each collection in example 1 so all future examples only use `Aggregate`?

Reply:
Sure! I moved the insertions for a set of examples to the first examples. I named groups of docs based on their collections (i.e. `sales`) and made separate results and errors (i.e. `salesResult` and `salesErr`) for each `InsertMany` that I still check outside of the example.